sound/soc/sof/topology.c

```diff
@@ -181,7 +187,14 @@ static int sof_keyword_dapm_event(struct snd_soc_dapm_widget *w,
 				swidget->widget->name);
 		break;
 	case SND_SOC_DAPM_POST_PMD:
-		if (spcm->stream[stream].suspend_ignored) {
+		/*
+		 * if the D0i3_pipeline_count is > 0, this indicates that
+		 * D0i3-compatible pipelines were not stopped during
+		 * suspend. So the KWD pipeline needs to be kept running
+		 * as well.
+		 */
+		if (spcm->stream[stream].d0i3_compatible &&
+		    sdev->D0i3_pipeline_count) {
```

Comment:
the logic here is becoming convoluted?

Reply:
How? There are 2 checks. How is it convoluted? In fact the first one could possibly be removed because the KWD stream is meant to be D0i3 compatible

Reply:
um, the wording D0i3_pipeline_count make it uneasy to follow to me, as you know, D0I3 stream/pipeline can be not suspend related, how about using 'suspend_ignored_count'?

Reply:
@keyonjie "suspend_ignored" is what makes me uneasy. From an ALSA point of view, none of the PCM streams ignore suspend.They do get "suspended" but the DSP pipeline remains running.